modules/openapi-generator/src/main/resources/go-experimental/api.mustache

```diff
@@ -242,7 +242,7 @@ func (r api{{operationId}}Request) Execute() ({{#returnType}}{{{.}}}, {{/returnT
 	if r.ctx != nil {
 		// API Key Authentication
 		if auth, ok := r.ctx.Value(ContextAPIKeys).(map[string]APIKey); ok {
-			if auth, ok := auth["{{keyParamName}}"]; ok {
+			if auth, ok := auth["{{#vendorExtensions.x-lookup}}{{.}}{{/vendorExtensions.x-lookup}}{{^vendorExtensions.x-lookup}}{{name}}{{/vendorExtensions.x-lookup}}"]; ok {
```

Comment:
It would be good to document this vendor extension somewhere. Currently the vendor extensions are listed at https://github.com/OpenAPITools/openapi-generator/wiki/Vendor-Extensions, but it does not look like only owners can make a change (?). Here the vendor extension is buried in the template, so I think very few people would know this aliasing capability exists.
Relevant issue:
#4976

Reply:
@sebastien-rosset I agree we should document it. Personally I would move the wiki to `/docs` so it's editable by contributors.

Reply:
I saw the switch from `{{keyParamName}}` to `{{name}}`. Is `keyParamName` casuing issue?

Reply:
Another minor feedback is that when I saw `{{#vendorExtensions.x-lookup}}`, I really have no idea what `lookup` means in this case.

What about `{{#vendorExtensions.x-auth-id-alias}}` instead?

Reply:
@wing328 other languages are using `{{keyParamName}}` and - **yes** - `{{name}}` can cause problem if query param and header have the same name.

Reply:
I don't have any strong opinion about the name. At first I thought about it as key aliasing, but during the implementation I have changed my mind can call it `x-lookup` since it's more about from where the value is looked up.

Reply:
I like William's suggestion of `x-auth-id-alias`.

Also, it sounds like `{{name}}` should be changed back to `{{keyParamName}}`? Maybe it's usage in `{{^vendorExtensions.x-lookup}}{{name}}{{/vendorExtensions.x-lookup}}` was unintentional?

Reply:
- I have changed the `x-lookup` to `x-auth-id-alias`.
- using `keyParamName` is wrong since there can be a collision when using API key in header and query argument under same name: `curl -H 'my-key: header-key' https://example.com/endpoint` vs. `curl https://example.com/endpoint?my-key=query-key`